ios/RCCNavigationController.m

```diff
@@ -272,10 +274,27 @@ -(void)setButtons:(NSArray*)buttons viewController:(UIViewController*)viewContro
     NSString *title = button[@"title"];
     UIImage *iconImage = nil;
     id icon = button[@"icon"];
+    NSString *customView = button[@"customView"];
     if (icon) iconImage = [RCTConvert UIImage:icon];
-    
+    NSString *buttonId = button[@"id"];
+    NSString *callbackId = button[@"onPress"];
+
     UIBarButtonItem *barButtonItem;
-    if (iconImage)
+    if (customView) {
+      RCTBridge *bridge = ((RCTRootView*)viewController.view).bridge;
+
+      NSNumber *viewWidth = button[@"customViewWidth"] ?: @60;
+      NSMutableDictionary *initialProps = [NSMutableDictionary dictionaryWithDictionary:button[@"customViewInitialProps"]];
+      initialProps[@"buttonID"] = buttonId;
+      initialProps[@"callbackID"] = callbackId;
+      RCTRootView *reactView = [[RCTRootView alloc] initWithBridge:bridge moduleName:customView initialProperties:initialProps];
+      CGRect buttonFrame = CGRectMake(0, 0, viewWidth.floatValue, self.navigationBar.bounds.size.height);
```

Comment:
Not every button needs to be as tall as the nav bar. Should we be relying on the view to set its own size and layout the RCCCustomButtonView relative to that?